### ISSUE TITLE
Error 401 when login

### ISSUE THREAD

**Issue by vuongminh** [luyadev/luya#1161]
Today i am up my project to **cpanel hosting** and i get an error when login user admin. **Login is success but redirect to logout** because it have an error in route **api-admin-menu** (error 401)

![untitled](https://cloud.githubusercontent.com/assets/13974585/21938986/4a878ef6-d9f0-11e6-8308-8a9efc1570dd.png)

**Comment by nadar** [comment]
401 is not authorized.

1. Have you grant the permissions to this user? Does the group has all permissions?
2. Its an apache2 webeserver? Is mod rewrite enabled on this host? 
3. Run this script and post the response please: https://gist.github.com/nadar/022f890e064c068e75ce6178cd14a09e

(btw: This "dashboard" (?) looks great!)

**Comment by vuongminh** [comment]
Thank you :)! 

My test script result:

```
1: [ini_get('short_open_tag')] '1'
2: [ini_get('error_reporting')] '32767'
3: [phpversion()] '7.0.14'
4: [php_ini_loaded_file()] '/opt/alt/php70/etc/php.ini'
5: [php_sapi_name()] 'cgi-fcgi'
```

I have export mysql database and import it to host => admin user has all permissions like my local app

**Comment by nadar** [comment]
Could you post a screenshot from the request, so i can see the `Authorization:Bearer ` from the request headers.

Like this from google chrome's developer toolbar Network Tab:
<img width="1236" alt="bildschirmfoto 2017-01-14 um 08 50 22" src="https://cloud.githubusercontent.com/assets/3417221/21953414/84d68b98-da36-11e6-9b24-86cc61fb359e.png">

**Comment by nadar** [comment]
(a dump of $_SERVER would be interesting too)

**Comment by vuongminh** [comment]
My request header:

![untitled](https://cloud.githubusercontent.com/assets/13974585/21954412/85e600da-da82-11e6-82bc-1671bb7a25bb.png)

dump of $_SERVER in login page or api-admin-menu? Because now i just can go to api-admin-menu with access-token param get.

Thank you for help! :D

**Comment by vuongminh** [comment]
When I login why it can authorized controllers but apis is not?

**Comment by nadar** [comment]
Cause controllers are controlled by the session, but apis are authorized by the bearer auth token which is available (as you can see from your screenshot) but your webserver seems not able to handle. Just print the $_SERVER does not matter where - i just want to check the `$_SERVER['SERVER_SOFTWARE']` are you sure its apache2 webserver?

**Comment by nadar** [comment]
Maybe try this: http://stackoverflow.com/questions/22562034/header-is-received-by-apache-but-not-present-in-php

**Comment by vuongminh** [comment]
My server software is Apache

```
 ["SERVER_SOFTWARE"]=>
  string(6) "Apache"
```

**Comment by nadar** [comment]
Have you tested? http://stackoverflow.com/a/22562390/4611030
What does your .htaccess look like?

**Comment by vuongminh** [comment]
I had test! But it not work! 

My htaccess:

```
Options +FollowSymLinks
RewriteEngine on

RewriteCond %{REQUEST_FILENAME} !-f
RewriteCond %{REQUEST_FILENAME} !-d

RewriteRule . index.php
```

**Comment by nadar** [comment]
Try this (which should be default by any LUYA project):

https://github.com/luyadev/luya/blob/master/envs/kickstarter/public_html/.htaccess#L5

**Comment by vuongminh** [comment]
I had use this htaccess but not work for me! My problem had found because in PHP not exist AUTHORIZATION header property but i don't know how to solved it :(

**Comment by nadar** [comment]
I know that problem (https://github.com/yiisoft/yii2/issues/6631) but it should be fixed with the htaccess rule. So i would say its related to your webserver if `var_dump($request->getHeaders()->get('Authorization'));` is empty.

Maybe drop me a message on gitter (https://gitter.im/nadar) so we can chat.

**Comment by vuongminh** [comment]
I had move my host to plesk. 

Thank you very much @nadar

**Comment by nadar** [comment]
@vuongminh I still can't believe its related to cpanel, maybe i can get a cpanel hosting somewhere in order to test. As there should be no big differents between those. But glad to hear it works now.

**Comment by PreneeshAV** [comment]
i faced a similar problem for a fresh installation on cpanel server with nginx web server. However i could resolved it by changing php default version to 7.0 and enabling **PHP-FPM** against the domain in WHM -  Home »Software »MultiPHP Manager
